src/app/components/handlers/usePasswordToggle.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState } from "react";
+
+const usePasswordToggle = () => {
```

Comment:
Rather than this function, wouldn't it make more sense to have a `Password` component? I'm guessing all the passwords on the website have the same functionality. So if this assumption is correct, then the clean way to do this would be with a `Password` component.

Reply:
Not entirely sure what you mean here; there are multiple password components through the site, and we want all of them to be controlled by a show/hide functionality, but only a single button. For example on the My Account page 

![Screenshot 2023-08-30 at 12 07 26](https://github.com/isaaccomputerscience/isaac-react-app/assets/114683476/b2a7a2ad-6a39-42c7-9d57-9e458a7e395d)

Reply:
Thanks for clarifying! However, what I meant is that it looks like the inputs that contain the eye icon should probably be a component by itself. At the moment they all look very similar, like this:
```jsx
<div className="password-group">
    <Input
        id="password-input" autoComplete="current-password" type={PasswordInputType} name="password" placeholder="Password"
        onChange={(event: React.ChangeEvent<HTMLInputElement>) => setPassword(event.target.value)}
        invalid={!!errorMessage || (!validPassword && (logInAttempted))}
        aria-describedby="passwordValidationMessage"
        required
    />
    <span className="password-toggle-icon" id="password-toggle">{ToggleIcon}</span>
</div>
```

So I would have expected all that (plus the `const [visible, setVisible] = useState(false);` logic) to be encapsulated in its own `PasswordInput` component, if that makes sense.

I think that the problem is that the original code was already quite convoluted. So what I'm asking is to evaluate how hard would it be to do it properly, instead of adding more complexity on top of something that wasn't very well executed in the first place.

Reply:
There are I think 7 instances of the password input, with differences between them that would need to be combined. That would be one for another ticket to fix as it would take a bit longer to cleanup the old code, or we'd need to delay adding the show/hide functionality until after the registration feature deploy. Definitely value to reducing the complexity!

I think even making a component just for the inputs that contain the eye icon wouldn't eliminate the need for the usePasswordToggle() because the inputs that don't have the icon still need their type controlled by the InputType